### PR TITLE
DevWs for NIE-TECH CO., LTD. containing Minoston Wallmote

### PR DESCRIPTION
**Pull request by sky-nie**

### CODE REVIEW

**Review by PKacprowiczS — CHANGES_REQUESTED** [review]
Check if you were editing most recent file version, cause you've deleted bunch of stuff from the DTH.

**Comment by sky-nie** [comment]
> Check if you were editing most recent file version, cause you've deleted bunch of stuff from the DTH

Oh，This is my mistake, not a modification based on the latest version. And we have change the DTH from [myself](https://raw.githubusercontent.com/sky-nie/SmartThingsPublic/master/devicetypes/sky-nie/minoston-wallmote.src/minoston-wallmote.groovy) to [the public](https://raw.githubusercontent.com/SmartThingsCommunity/SmartThingsPublic/master/devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy), as you suggested on [here](https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/70148)

In addition, I found that a fingerprint named "Aeotec illumino Wallmote 7" has been added into [the DTH](https://raw.githubusercontent.com/SmartThingsCommunity/SmartThingsPublic/master/devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy). The feature is that it adds support for "pushed_3x", "pushed_4x" and "pushed_5x". However, it uses vid: "generic-2-button" Obviously, this vid does not support the display of "pushed_3x", "pushed_4x" and "pushed_5x". Is this a bug?

**Comment by PKacprowiczS** [comment]
@sky-nie I believe that this device was added by Aeotec themselves and they were checking this out. Chances are that recent mobile app changes made `supportedButtonValues` event have higher priority over `vid` in terms of supported values. But I don't have the device at hand, so I'm not able to test it.

File: devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy [hunk 1]
```diff
@@ -11,9 +11,6 @@
  *  for the specific language governing permissions and limitations under the License.
  *
  */
-
-import groovy.json.JsonOutput
-
```
Comment (PKacprowiczS) [inline devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy]: This shouldn't be deleted.

File: devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy [hunk 2]
```diff
@@ -198,7 +196,7 @@ private getButtonAttributesMap() {
 }
 
 private isEverspring() {
-    zwaveInfo.model.equals("0003")
+    return (zwaveInfo.model == "0003" ||  zwaveInfo.model == "D001")
```
Comment (PKacprowiczS) [inline devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy]: Putting here your own model doesn't seem right, after all it isn't `Evepsring`.

File: devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy [hunk 3]
```diff
@@ -45,7 +43,7 @@ metadata {
 }
 
 def getNumberOfButtons() {
-    def modelToButtons = ["0082" : 4, "0081": 2, "0003": 2, "0016": 2]
+    def modelToButtons = ["D001" : 4, "0082" : 4, "0081": 2, "0003": 2, "0016": 2]
```
Comment (PKacprowiczS) [inline devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy:49]: You shouldn't alter existing models here.

File: devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy [hunk 4]
```diff
@@ -27,6 +24,7 @@ metadata {
         fingerprint mfr: "0086", model: "0081", deviceJoinName: "Aeotec Remote Control", mnmn: "SmartThings", vid: "generic-2-button" //Aeotec Wallmote
         fingerprint mfr: "0060", model: "0003", deviceJoinName: "Everspring Remote Control", mnmn: "SmartThings", vid: "generic-2-button" //Everspring Wall Switch
         fingerprint mfr: "0371", model: "0016", deviceJoinName: "Aeotec Remote Control", mnmn: "SmartThings", vid: "generic-2-button" //Aeotec illumino Wallmote 7
+        fingerprint mfr: "0312", model: "D001", deviceJoinName: "S2 Remote Control Switch", mnmn: "SmartThings", vid: "generic-4-button" //Minoston Wallmote
```
Comment (PKacprowiczS) [inline devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy]: change `deviceJoinName` to `Minoston Remote Control`.

**Review by PKacprowiczS — CHANGES_REQUESTED** [review]

File: devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy (hunk 3 above)
Comment (PKacprowiczS) [reply devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy:49]: Nevermind, it appears I had a wrong look here.

File: devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy (hunk 1 above)
Comment (sky-nie) [reply devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy]: OK, I restore it.

File: devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy (hunk 2 above)
Comment (sky-nie) [reply devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy]: OK, I am independent as a function.

File: devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy (hunk 4 above)
Comment (sky-nie) [reply devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy]: OK, I modify it.

File: devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy [hunk 5]
```diff
@@ -184,7 +187,12 @@ private getButtonAttributesMap() {
             0: "pushed",
             2: "held",
             3: "double"
-    ]} else if (isWallMote7()) {[
+    ]} else if (isMinoston()) {[
+            0: "pushed",
+            2: "held",
+            3: "double",
+            4: "pushed_3x"
+    ]}else if (isWallMote7()) {[
```
Comment (PKacprowiczS) [inline devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy]: space between `}` and `else`

File: devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy [hunk 6]
```diff
@@ -172,7 +173,9 @@ def getChildDevice(button) {
 private getSupportedButtonValues() {
     if (isEverspring()) {
         return ["pushed", "held", "double"]
-    } else if (isWallMote7()) {
+    } else if (isMinoston()) {
+        return ["pushed", "held", "double", "pushed_3x"]
+    }else if (isWallMote7()) {
```
Comment (PKacprowiczS) [inline devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy]: space between `}` and `else`

**Review by PKacprowiczS — CHANGES_REQUESTED** [review]

File: devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy (hunk 6 above)
Comment (sky-nie) [reply devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy]: OK, I modify it.

File: devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy (hunk 5 above)
Comment (sky-nie) [reply devicetypes/smartthings/aeotec-wallmote.src/aeotec-wallmote.groovy]: OK, I modify it.

**Review by PKacprowiczS — APPROVED** [review]
@sky-nie ok, it looks good, but why don't you edit PR's title and change `Aeotec` to `Wallmote` so it won't end up with confusing commit message
@greens

**Comment by sky-nie** [comment]
> @sky-nie ok, it looks good, but why don't you edit PR's title and change `Aeotec` to `Wallmote` so it won't end up with confusing commit message
> @greens
oh，I didn't know until you mentioned it now that this title can be modified.I always thought that the automatically generated title is unchangeable.

**Comment by PKacprowiczS** [comment]
@sky-nie my mistake in previous message, I obviously meant "change `Aeotec` to `Minoston`", but I spotted just now. Sorry for misleading you.

**Comment by sky-nie** [comment]
> @sky-nie my mistake in previous message, I obviously meant "change `Aeotec` to `Minoston`", but I spotted just now. Sorry for misleading you.

There should be no strict requirements for the PR title, right? I did not modify the name of DTH, because it is now modified on the basis of the official DTH, so the name cannot be modified.

**Comment by PKacprowiczS** [comment]
@sky-nie yeah, but current title indicates that PR adds a new Aeotec device, while in fact it adds Minoston. It's kinda misleading.

**Review by KKlimczukS — APPROVED** [review]

**Pull request merged**